pkg/kv/kvserver/intent_resolver_integration_test.go

```diff
@@ -164,3 +170,34 @@ func TestContendedIntentWithDependencyCycle(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// Regression test for https://github.com/cockroachdb/cockroach/issues/64092
+// which makes sure that synchronous ranged intent resolution during rollback
+// completes in a reasonable time.
+func TestRollbackSyncRangedIntentResolution(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	skip.UnderRace(t)
+	skip.UnderStress(t)
+
+	ctx := context.Background()
+	srv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			Store: &StoreTestingKnobs{
+				DisableLoadBasedSplitting: true,
+				IntentResolverKnobs: kvserverbase.IntentResolverTestingKnobs{
+					ForceSyncIntentResolution: true,
+				},
+			},
+		},
+	})
+	defer srv.Stopper().Stop(ctx)
+
+	txn := srv.DB().NewTxn(ctx, "test")
+	for i := 0; i < 100000; i++ {
+		require.NoError(t, txn.Put(ctx, []byte(fmt.Sprintf("key%v", i)), []byte("value")))
+	}
+	ctx, cancel := context.WithTimeout(ctx, 20*time.Second)
+	defer cancel()
+	require.NoError(t, txn.Rollback(ctx))
+}
```

Comment:
`require.NoError(t, ctx.Err())`

Reply:
That shouldn't be necessary, since `txn.Rollback(ctx)` will return the context error if and when it expires.

Reply:
Hm, although I suppose that assumes that the code will actually check the context, which all code may not do. Will add a check, thanks.

Reply:
Yeah this is just Defensive Programming(TM)